### DISCUSSION TITLE
Implementing Filtering via dropdown

### DISCUSSION THREAD

**Discussion by normanmadrid** [carbon-design-system/carbon-components-svelte#1617] [Q&A]
disclaimer: I just started developing with svelte a week ago
I'm trying to implement  a filter using the carbon dropdown component, I thought I could do this by dispatching an event on select, then going through the rows data with a conditional, when the conditional activates (filter applied) i add the ID of that row to filteredRowIds which is bound to my search bar. the problem I seem to be facing is even tho I add that row's ID to the filteredRowIds its not updating the table, some help would be very much appreciated, thanks!

**Comment by metonym** [comment]
Currently, the `filteredRowIds` prop on `ToolbarSearch` only supports one-way binding. Consequently, `filteredRowIds` cannot be updated programmatically.

One workaround is to provide the selected dropdown value to `ToolbarSearch` and utilize the built-in filtering logic.

[Svelte REPL](https://svelte.dev/repl/87d4501af18c44f69254f0db3a51d57d?version=3.55.0)

```svelte
<script>
  import {
    Dropdown,
    DataTable,
    Toolbar,
    ToolbarContent,
    ToolbarSearch,
  } from "carbon-components-svelte";

  let rows = Array.from({ length: 10 }).map((_, i) => ({
    id: i,
    name: "Load Balancer " + (i + 1),
    protocol: "HTTP",
    port: 3000 + i * 10,
    rule: i % 2 ? "Round robin" : "DNS delegation",
  }));
  let selectedId = "1";

  const items = [
    { id: "1", text: "Load Balancer 1" },
    { id: "2", text: "Load Balancer 2" },
  ];

  $: value = items.find((item) => item.id === selectedId).text;
</script>

<Dropdown bind:selectedId {items} />

<br />

<DataTable
  headers={[
    { key: "name", value: "Name" },
    { key: "protocol", value: "Protocol" },
    { key: "port", value: "Port" },
    { key: "rule", value: "Rule" },
  ]}
  {rows}
>
  <Toolbar>
    <ToolbarContent>
      <ToolbarSearch persistent shouldFilterRows bind:value />
    </ToolbarContent>
  </Toolbar>
</DataTable>

```

**Reply by normanmadrid** [reply]
damn thats a shame, i cant use this implementation because this only works for one dropdown, was hoping to stack multiple filters kinda of like here: https://svelte.dev/repl/8b2264e66d9a456abeb5f2b9322605f7?version=3.55.0
